### ISSUE TITLE
Remove surf and replace it with reqwest

### ISSUE THREAD

**Issue by KolbyML** [ethereum/trin#1446]
https://github.com/ethereum/trin/issues/1444 Risc-V support is blocked by surf https://github.com/http-rs/surf/issues/352#issuecomment-1709404352 surf has been unmaintained for the last 2 years. 

It seems like the 2 best options are 
- hyper
- reqwest

Apparently reqwest is built on hyper, with some extra utils which is nice

**Comment by morph-dev** [comment]
I just want to highlight that we already depend on both hyper (in [rpc](https://github.com/ethereum/trin/blob/ba63fb289c889a8ef8cfd63838633377adcde3f7/rpc/Cargo.toml#L19) and [ethportal-peertest](https://github.com/ethereum/trin/blob/ba63fb289c889a8ef8cfd63838633377adcde3f7/ethportal-peertest/Cargo.toml#L22) ) and reqwest (in [light-client](https://github.com/ethereum/trin/blob/ba63fb289c889a8ef8cfd63838633377adcde3f7/light-client/Cargo.toml#L25)) in our repo.

I think #1448 (replace surf with request) can go as is, but I think we should create separate PR to migrate hyper to reqwest as well.
